gnosis_vpn-lib/src/connection/destination_health.rs

```diff
@@ -0,0 +1,296 @@
+                _ => self.health.clone(),
+            },
+            // desired peer address becomes peered
+            Need::Peering(addr) if addresses.contains(&addr) => Health::ReadyToConnect,
+            // peered address lost its peer
+            Need::Peering(_) => Health::NotPeered,
+            // no peer available, even any channel lost its peer
+            Need::AnyChannel if addresses.is_empty() => match self.health {
+                Health::ReadyToConnect => Health::MissingPeeredChannel,
+                Health::MissingFundedChannel => Health::MissingPeeredFundedChannel,
+                _ => self.health.clone(),
+            },
+            // any peer will suffice for any channel need
+            Need::AnyChannel => match self.health {
+                Health::MissingPeeredChannel => Health::ReadyToConnect,
+                Health::MissingPeeredFundedChannel => Health::MissingFundedChannel,
+                _ => self.health.clone(),
+            },
+            Need::Nothing => self.health.clone(),
+        };
+        Self {
+            health,
+            need: self.need.clone(),
+            last_error: self.last_error.clone(),
+        }
+    }
+
+    pub fn channel_funded(&self, address: Address) -> Self {
+        let health = match self.need {
+            // needed channel becomes funded
+            Need::Channel(addr) if addr == address => match self.health {
+                Health::MissingFundedChannel => Health::ReadyToConnect,
+                Health::MissingPeeredFundedChannel => Health::MissingPeeredChannel,
+                _ => self.health.clone(),
+            },
+            // any channel becomes funded
+            Need::AnyChannel => match self.health {
+                Health::MissingFundedChannel => Health::ReadyToConnect,
+                Health::MissingPeeredFundedChannel => Health::MissingPeeredChannel,
+                _ => self.health.clone(),
+            },
+            _ => self.health.clone(),
+        };
+        Self {
+            health,
+            need: self.need.clone(),
+            last_error: self.last_error.clone(),
+        }
+    }
+
+    pub fn needs_peer(&self) -> bool {
+        match self.need {
+            Need::Channel(_) | Need::AnyChannel => matches!(
+                self.health,
+                Health::MissingPeeredChannel | Health::MissingPeeredFundedChannel | Health::NotPeered
+            ),
+            Need::Peering(_) => matches!(self.health, Health::NotPeered),
+            Need::Nothing => false,
+        }
+    }
+
+    pub fn needs_channel_funding(&self) -> Option<Address> {
+        match self.need {
+            Need::Channel(addr) => match self.health {
+                Health::MissingFundedChannel | Health::MissingPeeredFundedChannel => Some(addr),
+                _ => None,
+            },
+            _ => None,
+        }
+    }
+
+    pub fn is_ready_to_connect(&self) -> bool {
```

Comment:
```suggestion
    #[inline]
    pub fn is_ready_to_connect(&self) -> bool {
```

This could also be a free standing function accepting `Health` or an impl on `Health` itself.